src/promptflow/promptflow/_sdk/_service/utils/utils.py

```diff
@@ -37,44 +44,59 @@ def wrapper(*args, **kwargs):
 
 
 def get_port_from_config(create_if_not_exists=False):
-    (HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE).touch(mode=read_write_by_user(), exist_ok=True)
-    with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "r", encoding=DEFAULT_ENCODING) as f:
-        service_config = load_yaml(f) or {}
-        port = service_config.get(sys.executable, {}).get("port", None)
-    if not port and create_if_not_exists:
-        with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "w", encoding=DEFAULT_ENCODING) as f:
-            # Set random port to ~/.promptflow/pf.yaml
-            port = get_random_port()
-            service_config[sys.executable] = service_config.get(sys.executable, {})
-            service_config[sys.executable]["port"] = port
-            dump_yaml(service_config, f)
-    return port
-
-
-def kill_deprecated_service():
-    (HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE).touch(mode=read_write_by_user(), exist_ok=True)
-    with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "r", encoding=DEFAULT_ENCODING) as f:
-        service_config = load_yaml(f) or {}
-        port = service_config.get("service", {}).get("port", None)
-        if port:
-            if is_port_in_use(port):
-                logger.debug(f"Kill the deprecated port {port} got from service key in thr pfs.port file.")
-                kill_exist_service(port)
-            # remove service key field
+    global pfs_port_lock
```

Comment:
when read we don't lock, only do lock when we want to write the file. Do a atomic get, merge, write back inside the lock.

Reply:
better to update the PR description to align it with the real code change

Reply:
have updated